### PR TITLE
Gimli v1.0

### PR DESCRIPTION
**Pull request by jordidj**
## PR description
Preparing `gimli` for release.

## New features
**Pylbo**
- When preparing Legolas data for use with AMRVAC through `gimli`, you can now request scaling the perturbation by energy percentage.
- Generate MPI-AMRVAC files from `gimli`

## Checklist
- [x] all tests pass (and for new features, new tests are added)
- [x] documentation has been updated (if applicable)

### CODE REVIEW

File: post_processing/pylbo/gimli/equilibrium.py [hunk 1]
```diff
@@ -150,33 +150,40 @@ def __init__(
...
+        self.v02, self.v03 = sp.sympify(v02), sp.sympify(v03)
+        self.T0 = sp.sympify(T0)
+        self.B02, self.B03 = sp.sympify(B02), sp.sympify(B03)
+
+        self.grid_spacing = sp.sympify(legolas_grid_spacing)
 
         self._dict_phys = {
             "resistivity": [
-                resistivity,
+                sp.sympify(resistivity),
```
Comment (nicolasbrughmans) [inline post_processing/pylbo/gimli/equilibrium.py:177]: In `gimli/legolas.py` on line 173, there is an additional `sympify` statement for all these physics types. Remove one of the two instances?

**Comment by jordidj** [comment]
To do: add `gimli` section on the website under "Detailed guide" (with examples) and link the relevant part on "Implementing a custom setup"

**Comment by jordidj** [comment]
**LEFT TO DO**
_Primary_
- [x] Add cylindrical BC (+ regularity condition) and heating/cooling to `mod_usr.t` generation in the `Amrvac` class
- [x] Support for `B0field` splitting
- [x] Review new documentation (@nicolasbrughmans)
- [x] Make MPI-AMRVAC 3.3 the norm for GIMLI
- [x] Move array loading subroutines in the generated `mod_usr.t` file to a dedicated Legolas module in the MPI-AMRVAC repository and load that in generated user files
- [ ] Support for no-slip BC

_Secondary_
- [x] rho/p splitting doesn't work in cylindrical coordinates. Unclear as of now if GIMLI or MPI-AMRVAC issue

_To check_
- [x] Gravity sign

After release: restructure cooling curves to have MPI-AMRVAC and Legolas pull data from a single set of tables

**Comment by nicolasbrughmans** [comment]
Hi! *blows away the dust on this PR* I think we're finally ready to merge this PR as all features have been added. Are there still features/docs/tests missing or things that could be improved? Including no-slip boundary conditions is still on the to-do list, but I would argue that's something to be added once it becomes relevant (the boundary conditions to obtain a steady equilibrium are already non-trivial, so adding a rotating boundary in the mix will only complicate that).

I'm also preparing a PR on reworking the cooling curve names/structure to match with MPI-AMRVAC. Once the Gimli PR has been merged into `develop`, Gimli can be modified to take those new names into account so that the heatcool coupling of Legolas to AMRVAC can be automatic to reduce the possible errors.

File: post_processing/pylbo/gimli/equilibrium.py (hunk 1 above)
Comment (jordidj) [reply post_processing/pylbo/gimli/equilibrium.py:177]: If I remember correctly, I added this because something was not working otherwise, but I never figured out why. If you want, you can try removing one but make sure to test everything properly.

File: post_processing/pylbo/gimli/legolas.py [hunk 2]
```diff
@@ -203,8 +233,77 @@ def _validate_config(self):
...
+                    if self.config[key] != self.equilibrium.heatcool[key]:
+                        pylboLogger.warning(
+                            f"Key '{key}' in heatcool overrides value in config. "
+                            "Using value from heatcool."
+                        )
+                self.config[key] = self.equilibrium.heatcool[key]
+
+        elif self.config.get("radiative_cooling", None) is not None:
+            self.equilibrium.heatcool = {
+                "force_thermal_balance": self.config.get("force_thermal_balance", False)
```
Comment (jordidj) [inline post_processing/pylbo/gimli/legolas.py]: Is the `force_thermal_balance` default in Legolas not True? If so, it should probably be True here too

File: post_processing/pylbo/gimli/legolas.py [hunk 3]
```diff
@@ -203,8 +233,77 @@ def _validate_config(self):
...
+            }
+            self.equilibrium.heatcool["cooling_curve"] = self.config.get(
+                "cooling_curve", None
+            )
+            self.equilibrium.heatcool["ncool"] = self.config.get("ncool", 4000)
+
+        if self.equilibrium._dict_phys["heating"][
+            0
```
Comment (jordidj) [inline post_processing/pylbo/gimli/legolas.py:271]: why index 0? Also below

File: src/equilibria/smod_equil_internal_kink_instability.f08 [hunk 4]
```diff
@@ -30,19 +30,20 @@
...
-      cte_p0 = 9.0_dp
+      cte_p0 = 3.0_dp
       alpha = 5.0_dp / a0
 
       k2 = 1.0_dp
       k3 = 0.16_dp * alpha
     end if ! LCOV_EXCL_STOP
 
-    a0 = settings%grid%get_grid_end()
+    a0 = settings%grid%get_grid_end() / 0.999999_dp
```
Comment (jordidj) [inline src/equilibria/smod_equil_internal_kink_instability.f08:47]: Why does this need to be implemented like this?

**Review by jordidj — COMMENTED** [review]
Looks ready to be merged!

File: post_processing/pylbo/gimli/legolas.py (hunk 3 above)
Comment (nicolasbrughmans) [reply post_processing/pylbo/gimli/legolas.py:271]: If the user doesn't specify a heating function in the `Equilibrium`, `_dict_phys` still has "heating" as a key, so we need to check whether there is any expression associated to it. See also `eq._dict_phys["gravity"][0]`.

File: src/equilibria/smod_equil_internal_kink_instability.f08 (hunk 4 above)
Comment (nicolasbrughmans) [reply src/equilibria/smod_equil_internal_kink_instability.f08:47]: The default settings weren't working properly, i.e., the temperature blows up at the outer boundary r2 (since density is zero there if a0=r2). Would it be more transparant to set `a0 = settings%grid%get_grid_end() * 1.000001_dp`?

File: post_processing/pylbo/gimli/legolas.py (hunk 2 above)
Comment (nicolasbrughmans) [reply post_processing/pylbo/gimli/legolas.py]: True. In fact, it makes more sense to have it as default in the `Equilibrium` object too, unless a user-specified heating function is given.

File: src/equilibria/smod_equil_internal_kink_instability.f08 (hunk 4 above)
Comment (jordidj) [reply src/equilibria/smod_equil_internal_kink_instability.f08:47]: No, this is fine. Thanks!
